cli/src/commands/parse.rs

```diff
@@ -113,108 +146,49 @@ pub fn parse_depfile(
 ) -> StdResult<ParsedLockfile, ParseError> {
     // Try and determine dependency file format.
     let path = path.into();
-    let format = find_depfile_format(&path, depfile_type);
-
-    // Attempt to parse with all known parsers as fallback.
-    let (format, lockfile) = match format {
-        Some(format) => format,
-        None => return Ok(try_get_packages(path, project_root)?),
+    let (format, path) = match find_depfile_format(&path, depfile_type) {
+        Some((format, Some(path))) => (Some(format), path),
+        Some((format, None)) => (Some(format), path),
+        None => (None, path),
     };
```

Comment:
The previous code tracked `path` and `lockfile` separately, but now we combine them. This technically has one minor drawback.

Previously, if the CLI was given a manifest file and it found a matching lockfile, it would switch to parsing the lockfile. But if that parsing failed, it would **fallback to lockfile generation** like so:

```
> echo "BAD" > Cargo.lock; phylum parse Cargo.toml >/dev/null
⚠️  "Cargo.toml" is not a lockfile, using "/Users/kyle/code/cli/Cargo.lock" instead
Generating lockfile for manifest "Cargo.toml" using Cargo…
```

Now the code completely switches to `Cargo.lock` and fails if the file is invalid with no way to fallback to the originally provided file.

I don't know if this fallback was actually useful or not... But it's noteworthy that we have lost it now.

Reply:
This is how master behaves too. So it's at least not a regression from the status quo.

Reply:
This is not at all how `main` behaves... Take this sample Rust project:

```toml
[package]
name = "tmp"
version = "0.1.0"

[dependencies]
log = "0.4.20"
```

Overwrite `Cargo.lock` and then running `phylum parse` on `Cargo.toml` looks like this:

### `main`

```
> echo "BAD" > Cargo.lock; ~/code/cli/target/debug/phylum parse Cargo.toml
⚠️  "Cargo.toml" is not a lockfile, using "/Users/kyle/tmp/Cargo.lock" instead
Generating lockfile for manifest "Cargo.toml" using Cargo…
[
  {
    "name": "log",
    "version": "0.4.20",
    "type": "cargo",
    "lockfile": "Cargo.toml"
  }
]                                                                                                                                                                                                                                                                                                                               

```

### `move_parse`

```
> echo "BAD" > Cargo.lock; ~/code/cli/target/debug/phylum parse Cargo.toml
⚠️  "Cargo.toml" is not a lockfile, using "/Users/kyle/tmp/Cargo.lock" instead
❗ Error: could not parse dependency file "Cargo.toml"

Caused by:
    0: Dependency file parsing failed
    1: subprocess failed:
       ❗ Config access denied: Operation not permitted (os error 1)
       [2023-12-21T19:18:02Z ERROR phylum] Failed to save config: Operation not permitted (os error 1)
       ---------------- Update Available ----------------

       A new version of the Phylum CLI is available. Run

       	phylum update

       to update to the latest version!

       --------------------------------------------------


       Generating lockfile for manifest "Cargo.lock" using Cargo…
       ❗ Error: Lockfile generation failed! For details, see: https://docs.phylum.io/docs/lockfile_generation

       Caused by:
           package manager quit unexpectedly (code: Some(101)):

           error: the manifest-path must be a path to a Cargo.toml file
```

Reply:
Yeah my bad, I tested with the wrong version.